src/ot_orchestration/dynamic_dags.py

```diff
@@ -0,0 +1,155 @@
+"""Generic genetics DAG with batch job support."""
+
+from ot_orchestration.operators.manifest_operators import (
+    ManifestFilterOperator,
+    ManifestGenerateOperator,
+    ManifestReadOperator,
+    ManifestSaveOperator,
+    ManifestSubmitBatchJobOperator,
+)
+
+from airflow.models.baseoperator import chain
+
+import time
+from airflow.decorators import task
+import logging
+from airflow.models.taskinstance import TaskInstance
+from airflow.operators.python import get_current_context
+from airflow.exceptions import AirflowSkipException
+from airflow.utils.trigger_rule import TriggerRule
+from ot_orchestration.types import Manifest_Object
+from ot_orchestration.utils import IOManager
+
+
+@task(task_id="end")
+def end():
+    """Finish the DAG execution."""
+    logging.info("FINISHED")
+
+
+@task(
+    task_id="consolidate_manifests",
+    trigger_rule=TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS,
+)
+def consolidate_manifests(ti: TaskInstance | None = None) -> list[Manifest_Object]:
+    """Consolidate manifests from the execution mode branching."""
+    params = get_current_context().get("params")
+    if params["mode"] == "CONTINUE":
+        branch = "generate_staging_output"
+    else:
+        branch = "filter_failed_manifests"
+    manifests = ti.xcom_pull(branch)
+    if len(manifests) == 0:
+        raise AirflowSkipException("No manifests detected, skipping.")
+    return manifests
+
+
+@task.branch(task_id="begin")
+def begin() -> str:
+    """Start the DAG execution by choosing the execution mode."""
+    logging.info("START")
+    params = get_current_context().get("params")
+    logging.info(params)
+    if params["mode"] == "CONTINUE":
+        return "generate_manifests"
+    else:
+        return "read_existing_manifests"
+
+
+@task(task_id="collect_task_outcome", multiple_outputs=True)
+def collect_task_outcome(manifests: list[Manifest_Object]):
+    """Collect the task(s) outcome and return failed and succeded manifests."""
+    # we need to re-read the manifests to report the updated status of the tasks
+    manifest_paths = [m["manifestPath"] for m in manifests]
+    new_manifests: list[Manifest_Object] = IOManager().load_many(manifest_paths)
+
+    failed_manifests = []
+    succeded_manifests = []
+    status_flag_prefix = "pass"
+    for new_manifest in new_manifests:
+        # we assume that the initial state is success
+        succeded = True
+        for key, val in new_manifest.items():
+            if key.startswith(status_flag_prefix) and not val:
+                succeded = False
+
+        if succeded:
+            succeded_manifests.append(new_manifest)
+        else:
+            failed_manifests.append(new_manifest)
```

Comment:
```suggestion
    failed_manifests = []
    succeded_manifests = []
    for new_manifest in new_manifests:
        for key, val in new_manifest.items():
            if key.startswith("pass") and not val:
                failed_manifests.append(new_manifest)
                break
        else:
            succeded_manifests.append(new_manifest)
```

What do you think about this? It should produce the same result and it stops as soon as a failed item is found. Not sure how long those lists are tho.

Reply:
You are correct, my code would generate many duplicates. 😥  I must admit I will have to write tests for manifest handling.